src/BasicViewer.js

```diff
@@ -33,17 +35,32 @@ var BasicViewer = function(catTitle, features, container, fv) {
                         return 'up_pftv_feature up_pftv_' + d.type.toLowerCase();
                     })
                 ;
+                newShapes
```

Comment:
Had another look - "newShapes" variable is not required and makes the code more difficult to read for the person who hasn't written it.

See below:

```
                shapes.enter().append('path')
                    .attr('name', function(d, index) {
                        d.internalId = d.internalId === undefined ? catTitle + '_' + index : d.internalId;
                        return d.internalId;
                    })
                    .attr('class',function(d) {
                        return 'up_pftv_feature up_pftv_' + d.type.toLowerCase();
                    })
                    .filter(function(d) {
                        return d.color || Constants.getTrackInfo(d.type).color;
                    })
                    .style('fill', function(d) {
                        return d.color ? d.color
                            : Constants.getTrackInfo(d.type).color ? Constants.getTrackInfo(d.type).color
                            : 'black';
                    })
                    .style('stroke', function(d) {
                        return d.color ? d.color
                            : Constants.getTrackInfo(d.type).color ? Constants.getTrackInfo(d.type).color
                            : 'black';
                    })
                ;
```
You also need to change the following:
`                ViewerHelper.addEventsClassAndTitle(catTitle, shapes, fv, container);
`